paymentsheet/src/test/java/com/stripe/android/paymentelement/confirmation/lpms/foundations/LpmNetworkTestActivity.kt

```diff
@@ -0,0 +1,207 @@
+package com.stripe.android.paymentelement.confirmation.lpms.foundations
+
+import android.app.Application
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.createSavedStateHandle
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.CreationExtras
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.core.Logger
+import com.stripe.android.core.injection.ENABLE_LOGGING
+import com.stripe.android.core.injection.IOContext
+import com.stripe.android.core.injection.PUBLISHABLE_KEY
+import com.stripe.android.core.injection.STRIPE_ACCOUNT_ID
+import com.stripe.android.core.utils.requireApplication
+import com.stripe.android.paymentelement.callbacks.PaymentElementCallbackIdentifier
+import com.stripe.android.paymentelement.confirmation.ALLOWS_MANUAL_CONFIRMATION
+import com.stripe.android.paymentelement.confirmation.ConfirmationHandler
+import com.stripe.android.paymentelement.confirmation.injection.DefaultConfirmationModule
+import com.stripe.android.paymentelement.confirmation.lpms.foundations.network.StripeNetworkTestClient
+import com.stripe.android.payments.core.analytics.ErrorReporter
+import com.stripe.android.payments.core.injection.PRODUCT_USAGE
+import com.stripe.android.payments.core.injection.STATUS_BAR_COLOR
+import com.stripe.android.payments.core.injection.StripeRepositoryModule
+import com.stripe.android.testing.FakeErrorReporter
+import com.stripe.android.testing.FakeLogger
+import com.stripe.android.view.ActivityStarter
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.plus
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.parcelize.Parcelize
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+
+internal class LpmNetworkTestActivity : AppCompatActivity() {
+    val viewModel: TestViewModel by viewModels {
+        TestViewModel.Factory {
+            requireNotNull(Args.fromIntent(intent))
+        }
+    }
+
+    val confirmationHandler by lazy {
+        viewModel.confirmationHandler
+    }
+
+    val testClient by lazy {
+        viewModel.testClient
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        confirmationHandler.register(activityResultCaller = this, lifecycleOwner = this)
+    }
+
+    class TestViewModel @Inject constructor(
```

Comment:
Should we change this to only store the component in the view model? https://trailhead.corp.stripe.com/docs/mobile-sdk/android-team/android-activity-architecture#solution